### PR TITLE
build: Remove java classpath conventions

### PR DESCRIPTION
**Pull request by devinrsmith**
This removes our custom configuration hierarchy, historically inherited from the "fishlib" days.

Follow-up to #5683

Note: unlike #5683, this migration does not aim to preserve the exact configuration dependencies that were inherited in the java-classpath-conventions hierarchy. The configuration conventions were overly broad, and didn't represent the real dependencies that each project has. As part of this, our dependency on `org.jetbrains:annotations` changed from an implementation dependency (inherited through the `fishBase` configuration) to a compileOnly dependency to better reflect how we actually depend on it. We might expect that the jetbrains annotations would be removed from our application runtime classpath entirely, but that is not the case - it turns out that one of our transitive dependencies has a runtime dependency on it:

```
org.jetbrains:annotations:13.0                                                                                                                                                                                       
\--- org.jetbrains.kotlin:kotlin-stdlib:1.9.10                                                                                                                                                                       
     +--- io.confluent:kafka-protobuf-provider:7.6.0                                                                                                                                                                 
     |    \--- io.confluent:kafka-protobuf-serializer:7.6.0                                                                                                                                                          
     |         \--- project :extensions-kafka                                                                                                                                                                        
     |              \--- project :server                                                                                                                                                                             
     |                   \--- project :server-jetty                                                                                                                                                                  
     |                        \--- runtimeClasspath
```

In this case, it has the visible effect of "downgrading" our runtime classpath from what was previously `org.jetbrains:annotations:24.0.0` to `org.jetbrains:annotations:13.0`. Arguably, this is the more correct runtime dependency to provide given it matches what the exact version the transitive library requests.

### CODE REVIEW

File: Configuration/build.gradle [hunk 1]
```diff
@@ -16,6 +19,7 @@ dependencies {
     annotationProcessor libs.autoservice.compiler
 
     testImplementation project(path: ':Base', configuration: 'tests')
+    testImplementation libs.commons.compress
```
Comment (rcaudy) [inline Configuration/build.gradle:22]: This seems surprising. Weird thing to only need for test...

File: log-factory/sinks/log-to-logbuffer/build.gradle [hunk 2]
```diff
@@ -3,11 +3,8 @@ plugins {
   id 'io.deephaven.project.register'
 }
 
-configurations {
-  compileOnly.extendsFrom fishIo
-}
-
 dependencies {
+  compileOnly project(':Base')
```
Comment (rcaudy) [inline log-factory/sinks/log-to-logbuffer/build.gradle:7]: These `compileOnly` deps in the log-factory/X modules kind of surprise me.

File: Configuration/build.gradle (hunk 1 above)
Comment (devinrsmith) [reply Configuration/build.gradle:22]: Some of the tests depend on IOUtils:

```
> Task :Configuration:compileTestJava FAILED
/home/devin/dev/deephaven/deephaven-core/Configuration/src/test/java/io/deephaven/configuration/PropertyInputStreamLoaderTraditionalTest.java:14: error: package org.apache.commons.compress.utils does not exist
import org.apache.commons.compress.utils.IOUtils;
                                        ^
/home/devin/dev/deephaven/deephaven-core/Configuration/src/test/java/io/deephaven/configuration/PropertyInputStreamLoaderTraditionalTest.java:31: error: cannot find symbol
            bytes = IOUtils.toByteArray(in);
                    ^
  symbol:   variable IOUtils
  location: class PropertyInputStreamLoaderTraditionalTest
/home/devin/dev/deephaven/deephaven-core/Configuration/src/test/java/io/deephaven/configuration/PropertyInputStreamLoaderTraditionalTest.java:43: error: cannot find symbol
            bytes = IOUtils.toByteArray(in);
                    ^
  symbol:   variable IOUtils
  location: class PropertyInputStreamLoaderTraditionalTest
3 errors
```

File: log-factory/sinks/log-to-logbuffer/build.gradle (hunk 2 above)
Comment (devinrsmith) [reply log-factory/sinks/log-to-logbuffer/build.gradle:7]: I agree; I don't think there is a technical reason why they need to be `compileOnly` at this level. Originally, I think this was done as a handwavy way to say that 3rd party code could depend on the logging API without pulling in the libraries themselves in the context of a "plugin" (sort of like how we have setup the authentication shadow jar stuff). If a 3rd party needs to have those semantics, they can declare their dependency as compileOnly themselves; ie, `compileOnly 'io.deephaven:deephaven-log-to-logbuffer:X.Y.Z'` for example.

Comment (devinrsmith) [reply log-factory/sinks/log-to-logbuffer/build.gradle:7]: Filed https://github.com/deephaven/deephaven-core/issues/5732

**Review by rcaudy — APPROVED** [review]

**Pull request merged**